### PR TITLE
[cln] add libcln

### PR DESCRIPTION
**Pull request by endingly**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Thanks for your PR, please fix the build error on CI pipeline. If this port doesn't support these triplets, please add `supports` in `vcpkg.json`.
```
libcln:x64-uwp
libcln:arm-uwp
libcln:arm-neon-android
```

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by endingly** [comment]
> Thanks for your PR, please fix the build error on CI pipeline. If this port doesn't support these triplets, please add `supports` in `vcpkg.json`.
> 
> ```
> libcln:x64-uwp
> libcln:arm-uwp
> libcln:arm-neon-android
> ```

What is platform expression of `arm-neon-android` ?

File: ports/cln/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dg0yt) [inline ports/cln/portfile.cmake:1]: Is this a general limitation, or just the lack of Windows DLL export definitions?

File: ports/cln/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+)
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+vcpkg_copy_tools(
+    TOOL_NAMES pi
+    SEARCH_DIR ${CURRENT_PACKAGES_DIR}/bin
+    DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT}
```
Comment (dg0yt) [inline ports/cln/portfile.cmake:29]: I think these are the defaults. Other port hardly use `SEARCH_DIR` and `DESTINATION`

**Review by dg0yt — COMMENTED** [review]
IMO the usage should be trimmed to 
~~~cmake
cln provides CMake targets:

    find_package(cln CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cln::cln)

File: ports/cln/usage [hunk 3]
```diff
@@ -0,0 +1,14 @@
+The package cln can be used via CMake:
+
+    find_library(cln CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/cln/usage:3]:
```suggestion
    find_package(cln CONFIG REQUIRED)
```

File: ports/cln/portfile.cmake (hunk 1 above)
Comment (endingly) [reply ports/cln/portfile.cmake:1]: In fact, libcln provides dynamic libraries, but compiling and exporting can cause multiple definitions. Do you know how to easily solve this problem?

Comment (endingly) [reply ports/cln/portfile.cmake:1]: For Windows, if compiling dynamic libraries, two errors will occur: 
1. redefinition of symbol
2. unable to parse some symbols in the library

Comment (endingly) [reply ports/cln/portfile.cmake]: Okay, I gave up. I moved Windows out of the support list.

Comment (dg0yt) [reply ports/cln/portfile.cmake]:
~~~cmake
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**